components/map.jsx

```diff
@@ -2,25 +2,69 @@ var React = require('react');
 
 var Map = React.createClass({
 
-    propTypes: {
-      accessToken: React.PropTypes.string.isRequired,
-      mapId: React.PropTypes.string.isRequired,
-    },
-    componentDidMount: function () {
-      require('mapbox.js'); // this will automatically attach to Window object.
-      // Create a map in the div mapboxDiv element
-      this.map = L.mapbox.map(this.refs.mapboxDiv.getDOMNode(), this.props.mapId, {accessToken: this.props.accessToken});
-    },
-    componentWillUnmount: function() {
-      this.map.remove();
-    },
-    // Called on initialization and after each change to the components
-    // props or state
-    render: function () {
-      return (
-        <div ref="mapboxDiv"></div>
-      )
-    }
-  });
+  propTypes: {
+    accessToken: React.PropTypes.string.isRequired,
+    mapId: React.PropTypes.string.isRequired,
+    mapParentID: React.PropTypes.string.isRequired,
+    mapClassName: React.PropTypes.string.isRequired
+  },
+  componentDidMount: function() {
+    var that = this;
+    require('mapbox.js'); // this will automatically attach to Window object.
+    // We have to set this globally in order to make .featureLayer() to work
+    L.mapbox.accessToken = this.props.accessToken;
```

Comment:
Actually, it looks like it wasn't working earlier because `accessToken` needs to be passed as an option to `featureLayer` constructor too: https://www.mapbox.com/mapbox.js/api/v2.1.5/l-mapbox-featurelayer/

Could you change it to use that?